### DISCUSSION TITLE
windows平台的解码选择

### DISCUSSION THREAD

**Discussion by Lonely233233** [hooke007/mpv_PlayKit#293] [mpv 通用]
hwdec=auto-copy/d3d11va-copy/nvdec-copy
选择哪个更好一点

**Comment by hooke007** [comment]
faq里有写。

https://github.com/hooke007/MPV_lazy/discussions/255#discussioncomment-4685342

**Comment by dyphire** [comment]
Windows上解码方式优先d3d11va和nvdec-copy。单纯测试解码性能的参考度并不大，还要参考实际测试h264,vc1,hevc,vp8,vp9,av1等常见视频格式下的表现来选择

直接说结论：d3d11va模式具有最好的硬件解码支持。相比软解它可以正常解码av1 4k60帧，后者不够强劲的CPU会掉帧（测试为i7-11800H）；相比nvdec模式只有d3d11va实现了8k av1的硬解支持，可以做到正常播放而不掉帧（软解放弃治疗）。nvdec虽然有yuv444硬解支持，但这种格式相对少见且多为hevc软解上并无压力，没有理由使用nvdec模式

参考 https://github.com/dyphire/mpv-config/commit/e6f33b4b0903e59387687dda0b654bdb11b69481

**Reply by hooke007** [reply]
d3d11va-copy 不推荐，会有奇怪的bug。曾经是mpv-lazy默认的解码，但是问题实在出的很离奇，改为了稳定的dxva2-copy

**Reply by dyphire** [reply]
dxv2-copy更不推荐，我测试中遇到过多次解码异常画面丢失
d3d11va-copy暂时测试中没有发现问题，不过改为nvdec-copy也没问题，虽然解码性能略高，但非8k av1下表现和d3d11va-copy没有太大的差距，之前一直使用也没发现过问题；至于8k av1反正只有d3d11va能硬解不掉帧，copy模式哪个都一样

edit: copy模式改为nvdec-copy，确保最大兼容性避免可能存在的bug

**Reply by hooke007** [reply]
> dxv2-copy更不推荐，我测试中遇到过多次解码异常画面丢失

需要具体证明。

> 至于8k av1反正只有d3d11va能硬解不掉帧，copy模式哪个都一样

当前d3d11下native硬解也没得选

**Reply by dyphire** [reply]
跟d3d11没关系，vulkan下使用native硬解模式nvdec是不支持8k av1硬解的，应该反过来说非d3d11模式用不了d3d11va硬解才是真的没得选

**Reply by hooke007** [reply]
我说的不就是d3d11 api???

**Reply by hooke007** [reply]
vk的问题faq里也早提到了，我已经引导他去完整看faq了，这些肯定是没必要复述了

**Reply by dyphire** [reply]
> nvdec也能啊，就是显存占用明显高了很多，我这8GB跑10bit的会oom

即使启用hwdec-codecs=all，nvdec也无法解码8k av1（基于 https://github.com/shinchiro/mpv-winbuild-cmake ）

**Reply by dyphire** [reply]
40系？我的3060解不了

**Reply by dyphire** [reply]
那还是d3d11va的原生硬解加速实现的比较好，我这边用它硬解8k av1没有压力
等上游vulkan解码实现再测试下表现吧，印象中目前只实现了h264,hevc,av1的硬解支持，还不是很实用（虽然其他格式软解也没啥压力

**Comment by OmiChrome** [comment]
so what is best for non-gpu i mean intel CPU 
with ffmepg libav support?

**Reply by hooke007** [reply]
No GPU no hwdec. I dont understand what you were talking about.

**Reply by OmiChrome** [reply]
use filter in mpv with cpu hwdec

**Reply by hooke007** [reply]
The "H" in `Hwdec` means GPU.

**Reply by hooke007** [reply]
CPU decoding is called software decoding (swdec)

**Reply by OmiChrome** [reply]
what about iGPU?

**Reply by hooke007** [reply]
Unless you had a power-overwhelming iGPU, usually I think there is no need to care about it (hwdec=auto/auto-copy).
